controllers/component_dependency_update_controller.go

```diff
@@ -0,0 +1,484 @@
+		}
+		return ctrl.Result{}, err
+	}
+
+	component, err := GetComponentFromPipelineRun(r.Client, ctx, pipelineRun)
+	if err != nil || component == nil {
+		log.Error(err, "failed to get component")
+		return ctrl.Result{}, err
+	}
+
+	if len(component.Spec.BuildNudgesRef) == 0 {
+		log.Info(fmt.Sprintf("component %s has no BuildNudgesRef set", component.Name))
+		return ctrl.Result{}, nil
+	}
+
+	if pipelineRun.IsDone() || pipelineRun.DeletionTimestamp != nil {
+		if controllerutil.ContainsFinalizer(pipelineRun, NudgeFinalizer) || pipelineRun.Annotations == nil || pipelineRun.Annotations[NudgeProcessedAnnotationName] == "" {
+			// Pipeline run is done and we have not cleared the finalizer yet
+			// We need to perform our nudge
+			return r.handleCompletedBuild(ctx, pipelineRun, component)
+		}
+	} else if !controllerutil.ContainsFinalizer(pipelineRun, NudgeFinalizer) {
+		log.Info("adding finalizer")
+		// We add a finalizer to make sure we see the run before it is deleted
+		// As tekton results should aggressivly delete when pruning is enabled
+		controllerutil.AddFinalizer(pipelineRun, NudgeFinalizer)
+		err := r.Client.Update(ctx, pipelineRun)
+		if err != nil {
+			log.Error(err, "failed to add finalizer")
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+// handleCompletedBuild will perform a 'nudge' updating dependent downstream components
+// This will involve creating a PR updating their references to our images to the newly produced image
+// For now it just produces a log message
+func (r *ComponentDependencyUpdateReconciler) handleCompletedBuild(ctx context.Context, pipelineRun *tektonapi.PipelineRun, updatedComponent *applicationapi.Component) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx)
+	success := pipelineRun.Status.GetCondition(apis.ConditionSucceeded).IsTrue()
+	if !success {
+		log.Info("not performing nudge as pipeline failed")
+		return r.removePipelineFinalizer(ctx, pipelineRun)
+	}
+	// find the image and digest we want to update to
+	image := ""
+	tag := ""
+	digest := ""
+	for _, r := range pipelineRun.Status.Results {
+		if r.Name == ImageDigestParamName {
+			digest = r.Value.StringVal
+		} else if r.Name == ImageUrlParamName {
+			image = r.Value.StringVal
+		}
+	}
+	// Failure to find is a permanent error, so remove the finalizer
+	if image == "" {
+		log.Error(fmt.Errorf("unable to find %s param on PipelineRun, not performing nudge", ImageUrlParamName), "no image url result")
+		return r.removePipelineFinalizer(ctx, pipelineRun)
+	}
+	if digest == "" {
+		log.Error(fmt.Errorf("unable to find %s param on PipelineRun, not performing nudge", ImageDigestParamName), "no image digest result")
+		return r.removePipelineFinalizer(ctx, pipelineRun)
+	}
+	parts := strings.Split(image, ":")
+	if len(parts) != 2 {
+		log.Error(fmt.Errorf("invalid image name %s", image), "invalid image format")
+		return r.removePipelineFinalizer(ctx, pipelineRun)
+	}
+	tag = parts[1]
+	repo := parts[0]
```

Comment:
1. Does IMAGE_URL always include a tag?
2. Port numbers (e.g. `image-registry.openshift-image-registry.svc:5000`) could break this too, but that may be an acceptable bug for now

Reply:
I will change it to split manually lastIndexOf ':'. There should never be a digest as this is a separate field. In practice this is always redhat-user-workloads anyway.